build-and-deploy.sh

```diff
@@ -35,7 +35,18 @@ if [ "${TRAVIS_BRANCH}" = "master" ] && [ "${TRAVIS_PULL_REQUEST}" = "false" ];
     JEKYLL_ENV=production bundle exec jekyll build --baseurl="${BASE_URL}"
 
     # Publish
-    git clone -b ${TARGET_BRANCH} https://${GITHUB_PAT}@github.com/$TRAVIS_REPO_SLUG.git OUTPUT
+    if [ "${DEPLOY_KEY}" != "" ]; then
+        echo "Setting deploy key"
+        eval $(ssh-agent -s)
+        # Hack to make the key from the env var have real newlines
+        echo "${DEPLOY_KEY}" | sed -e 's/\\n/\
+/g' | ssh-add -
```

Comment:
Can we use `s/\\n/\n/g` here?

It works well with bash on my Debian GNU/Linux:

```bash
#!/bin/bash

DEPLOY_KEY="-----BEGIN\ OPENSSH\ PRIVATE\ KEY-----\\naaa\\nbbb"
echo "${DEPLOY_KEY}"
echo "${DEPLOY_KEY}" | sed -e 's/\\n/\n/g'
```

```console
$ /tmp/a.sh
-----BEGIN\ OPENSSH\ PRIVATE\ KEY-----\naaa\nbbb
-----BEGIN\ OPENSSH\ PRIVATE\ KEY-----
aaa
bbb
```

Reply:
Didn't work on my mac but it doesn't have to as long as it works on Travis ;)

I'll give it a shot.

Reply:
Seems to work.

Reply:
Thanks for the try!

Reply:
Thanks for always managing to teach me a new bash trick :)